cfgov/ask_cfpb/tests/test_search.py

```diff
@@ -1,445 +1,95 @@
+        assert english_answerPage.search_term == search_term
+        assert english_answerPage.base_query == None
+        assert english_answerPage.suggestion == None
+        assert english_answerPage.results == []
+
+
+    def test_AnswerPageSearch_init_es(self):
+        search_term = 'impuesto'
+        spanish_answerPage = AnswerPageSearch(search_term=search_term, language='es')
+        assert spanish_answerPage.language == 'es'
+        assert spanish_answerPage.search_term == search_term
+        assert spanish_answerPage.base_query == None
+        assert spanish_answerPage.suggestion == None
+        assert spanish_answerPage.results == []
+
+    def test_AnswerPageSearch_autocomplete(self):
+        test_answer_page = AnswerPage(title="Money 101", question="What is money?", slug="test-answer-page", live=True)
+        self.ROOT_PAGE.add_child(instance=test_answer_page)
+        self.rebuild_elasticsearch_index("ask_cfpb", stdout=StringIO())
+        search_term = "mone"
+        test_answer_page_search = AnswerPageSearch(search_term=search_term)
+        self.assertEqual(test_answer_page_search.autocomplete()[0]['question'], "What is money?")
+
+    def test_AnswerPageSearch_autocomplete_error(self):
+        with mock.patch("ask_cfpb.documents.AnswerPageDocument.search", side_effect=RequestError):
+            results = AnswerPageSearch.autocomplete('mone')
+            self.assertEqual(results, [])
+    
+    def test_AnswerPage_search(self):
+        test_answer_page = AnswerPage(title="Money 101", question="What is money?", answer_content="Money makes the world go round.", slug="test-answer-page", live=True)
+        self.ROOT_PAGE.add_child(instance=test_answer_page)
+        self.rebuild_elasticsearch_index("ask_cfpb", stdout=StringIO())
+        search_term = "What is money?"
+        test_answer_page_search = AnswerPageSearch(search_term=search_term)
+        test_answer_page_search_results = test_answer_page_search.search()['results'][0].text.rstrip()
+        self.assertEqual(test_answer_page_search_results, "What is money?")
+
+    def test_AnswerPage_suggest(self):
+        test_answer_page = AnswerPage(title="Money 101", question="What is money?", answer_content="Money makes the world go round.", slug="test-answer-page3", live=True)
+        self.ROOT_PAGE.add_child(instance=test_answer_page)
+        self.rebuild_elasticsearch_index("ask_cfpb", stdout=StringIO())
+        search_term = "monye"
+        test_answer_page_search = AnswerPageSearch(search_term=search_term)
+        test_answer_page_search_results = test_answer_page_search.suggest()
+        self.assertEqual(test_answer_page_search_results['search_term'], "money")
+
+    def test_AnswerPage_suggest_error(self):
+        with mock.patch("ask_cfpb.documents.AnswerPageDocument.search", side_effect=IndexError):
+            try:
+                AnswerPageSearch.suggest('money')
+            except IndexError:
+                assert True
```

Comment:
Here you could use [`assertRaises`](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaises):

```python
with self.assertRaises(IndexError):
    AnswerPageSearch.suggest("money")
```

However, there are a few things here — the way this is being called, `AnswerPageSearch.suggest("money")` is calling `suggest()` as if it were a class method, and providing `"money"` as the `self` argument. I think you'll need to instead call `AnswerPageSearch("money").suggest()`.

The other thing is, if this is intended to test the result of search.py lines 88-92, you'd want instead to check that the result looks like the `dict` it's supposed to return, with "suggestion" being `None`, if the `AnswerPageDocument .search` method raises an `IndexError` (which the mock is intended to do). So something like:

```python 
result = AnswerPageSearch("money").suggest()
self.assertIsNone(result["suggestion"])
```

I think this will get you to the coverage you're looking for.

Reply:
doooope! thanks for this; def was feeling defeated this morning with my various IndexError permutations